linkerd/app/outbound/src/http/logical/policy/route/retry.rs

```diff
@@ -158,3 +163,14 @@ impl RetryPolicy {
         false
     }
 }
+
+// === impl RetryLabelExtract ===
+
+impl<B> ExtractParam<RouteLabels, http::Request<B>> for RetryLabelExtract {
+    fn extract_param(&self, t: &http::Request<B>) -> RouteLabels {
+        let Self(parent, route) = self;
+        let host = t.uri().host();
+
+        RouteLabels::new(parent.clone(), route.clone(), host)
+    }
+}
```

Comment:
and here, we actually get the hostname, provided the `Service<T>` target: a request with some sort of body.